Automattic/woocommerce-payments
# Add additional checks for cart and checkout blocks for script handles

Fixes #8328

#### Changes proposed in this Pull Request


This fix just adds some conditionals to enqueuing the payment method script handles to account for pages that have the cart or checkout block on them. In #8259 we limited the enqueuing of scripts to only the relevant pages, but when testing a page that wasn't officially set as the cart page in WC Settings, it was discovered that some styles were missing.

This applied to both the WooPay button and Stripe Link button.

The fix was just to use the `has_block` check in addition to the `is_cart` or `is_checkout` checks.







#### Testing instructions



1. Use the latest build of WooPayments (or develop branch).
2. Enable WooPay and add the button to display on cart and checkout.
3. Add pages for cart and checkout blocks if they don't already exist.
4. Make sure that those pages are **not** set as the WC Cart and Checkout pages.
5. Add an item to the cart and go cart.
6. Confirm on the cart block page, the WooPay button renders correctly.
7. Confirm on the checkout block page, the WooPay button renders correctly.
8. In settings disable WooPay and enable Stripe Link.
9. Go to checkout block page and enter an email address and tab off email.
10. Confirm Stripe Link button is displayed properly.



**Before**
![Screenshot 2024-03-05 at 3 33 19 PM](https://github.com/Automattic/woocommerce-payments/assets/68524302/b95defd9-35b1-4452-88ba-a462cb4976cd)
![Screenshot 2024-03-05 at 3 33 27 PM](https://github.com/Automattic/woocommerce-payments/assets/68524302/45c520e1-2e2c-4cbc-87af-19bdac7bdead)


**After**
![Screenshot 2024-03-05 at 3 35 37 PM](https://github.com/Automattic/woocommerce-payments/assets/68524302/dfa9e5ee-ea1f-478e-8f40-1ca26e34430f)
![Screenshot 2024-03-05 at 3 35 29 PM](https://github.com/Automattic/woocommerce-payments/assets/68524302/2879e0df-1117-4d7b-9ada-b758f3ef0f56)

*

-------------------


**Post merge**
